src/Metal/MTLResourceStateCommandEncoder.cs

```diff
@@ -0,0 +1,38 @@
+//
+// API for the Metal framework
+//
+// Authors:
+//   Manuel de la Pena <mandel@microsoft.com>
+//
+#if IOS 
+using System;
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+
+using Foundation;
+using ObjCRuntime;
+
+namespace Metal {
+
+	public static partial class MTLResourceStateCommandEncoder_Extensions {
+
+		[NoMac, NoTV, iOS (13,0)]
+		public static void Update (this IMTLResourceStateCommandEncoder This, IMTLTexture texture, MTLSparseTextureMappingMode mode, MTLRegion[] regions, nuint[] mipLevels, nuint[] slices, nuint numRegions)
+		{
```

Comment:
`numRegions` should be used for validation *or* hidden if it's always the `regions.Length`